sql-statements/sql-statement-alter-table-compact.md

```diff
@@ -89,11 +89,108 @@ To obtain greater table-level concurrency with higher resource usage, you can mo
++------------------+-------------------+
+| TOTAL_DELTA_ROWS | TOTAL_STABLE_ROWS |
++------------------+-------------------+
+|                3 |                 0 |
++------------------+-------------------+
+-- Newly written data can be compacted
+ALTER TABLE foo COMPACT TIFLASH REPLICA;
+SELECT TOTAL_DELTA_ROWS, TOTAL_STABLE_ROWS FROM INFORMATION_SCHEMA.TIFLASH_TABLES
+    WHERE IS_TOMBSTONE = 0 AND
+    `TIDB_DATABASE` = "test" AND `TIDB_TABLE` = "foo";
++------------------+-------------------+
+| TOTAL_DELTA_ROWS | TOTAL_STABLE_ROWS |
++------------------+-------------------+
+|                0 |                 3 |
++------------------+-------------------+
+-- All data is in the best state and no compaction is required
+```
+
+</details>
+
+<details>
+  <summary>Example: Check the compaction state of partitioned tables</summary>
+
+```sql
+USE test;
+CREATE TABLE employees
+    (id INT NOT NULL, store_id INT)
+    PARTITION BY LIST (store_id) (
+        PARTITION pNorth VALUES IN (1, 2, 3, 4, 5),
+        PARTITION pEast VALUES IN (6, 7, 8, 9, 10),
+        PARTITION pWest VALUES IN (11, 12, 13, 14, 15),
+        PARTITION pCentral VALUES IN (16, 17, 18, 19, 20)
+    );
+ALTER TABLE employees SET TIFLASH REPLICA 1;
+INSERT INTO employees VALUES (1, 1), (6, 6), (10, 10);
+SELECT PARTITION_NAME, TOTAL_DELTA_ROWS, TOTAL_STABLE_ROWS
+    FROM INFORMATION_SCHEMA.TIFLASH_TABLES t, INFORMATION_SCHEMA.PARTITIONS p
+    WHERE t.IS_TOMBSTONE = 0 AND t.TABLE_ID = p.TIDB_PARTITION_ID AND
+    p.TABLE_SCHEMA = "test" AND p.TABLE_NAME = "employees";
++----------------+------------------+-------------------+
+| PARTITION_NAME | TOTAL_DELTA_ROWS | TOTAL_STABLE_ROWS |
++----------------+------------------+-------------------+
+| pNorth         |                1 |                 0 |
+| pEast          |                2 |                 0 |
+| pWest          |                0 |                 0 |
+| pCentral       |                0 |                 0 |
++----------------+------------------+-------------------+
+-- Some partitioned tables can be compacted
+ALTER TABLE employees COMPACT TIFLASH REPLICA;
+SELECT PARTITION_NAME, TOTAL_DELTA_ROWS, TOTAL_STABLE_ROWS
+    FROM INFORMATION_SCHEMA.TIFLASH_TABLES t, INFORMATION_SCHEMA.PARTITIONS p
+    WHERE t.IS_TOMBSTONE = 0 AND t.TABLE_ID = p.TIDB_PARTITION_ID AND
+    p.TABLE_SCHEMA = "test" AND p.TABLE_NAME = "employees";
++----------------+------------------+-------------------+
+| PARTITION_NAME | TOTAL_DELTA_ROWS | TOTAL_STABLE_ROWS |
++----------------+------------------+-------------------+
+| pNorth         |                0 |                 1 |
+| pEast          |                0 |                 2 |
+| pWest          |                0 |                 0 |
+| pCentral       |                0 |                 0 |
++----------------+------------------+-------------------+
+-- Data in al partitions is in the best state and no compaction is required
```

Comment:
```suggestion
-- Data in all partitions is in the best state and no compaction is needed
```

Compaction is an optional action so that I think using "needed" would be better.